sql_enum.gemspec

```diff
@@ -25,11 +25,10 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ["lib"]
 
-  spec.add_dependency "activesupport", ">= 6.0", "< 7.0"
-  spec.add_dependency "activerecord", ">= 6.0", "< 7.0"
```

Comment:
I think we probably still need at least a floor dependency for rails in here. Otherwise this may resolve for existing Rails 6.x codebases, and from what I can tell, you've removed that compat.

Possibly pinning to both? I know that makes upgrading to Rails 8 harder, but if it's actually compatible, that would be a small change.

```
spec.add_dependency "rails", "~> 7.0"
```

Reply:
It's pinned in the gemfile rather then the gemspec to `~> 7.0.0`

Reply:
Won't that mean someone could add this to a Rails 6 project and not get a error from bundler?

Reply:
I don't think so? I think ~> 7.0.0 means between 7.0.0 and 7.0.x?

Reply:
Yeah, Bundler is only looking at the `gemspec`. I agree it makes sense to set a floor here so people don't accidentally get tripped up.

Reply:
Bundler also looks at the gemfile, which is pinned to `~> 7.0.0`. It should be fine.

Reply:
I'm moderately certain that this isn't true. I tried locally linking the gem and the generated lockfile only shows the gemspec requirements:

<img width="514" alt="Screenshot 2024-01-26 at 8 05 35 AM" src="https://github.com/1debit/sql_enum/assets/97126321/39dad209-f953-483d-a3af-ed861374aed7">

It was also willing to resolve this on my Rails 6 install:

<img width="503" alt="Screenshot 2024-01-26 at 8 06 53 AM" src="https://github.com/1debit/sql_enum/assets/97126321/0956d621-2641-465b-8103-2b79c914648e">

Reply:
That's my mistake -- bundler looks at the gemfile when you are developing the gem, when you are linking the gem, it only looks at the gemspec. I'll fix it.

Reply:
(I switched it because I needed the gemfile to run git bisect over Rails when I was trying to verify when Rails behavior changed)